File: implementations/fastapi/examples/src/order/interface/rest/schemas.py
from pydantic import BaseModel


class ItemInputSchema(BaseModel):
    item_id: int
    name: str
    price: int
    quantity: int


class CreateOrderRequest(BaseModel):
    user_id: str
    items: list[ItemInputSchema]


class CancelOrderRequest(BaseModel):
    reason: str


class OrderSummaryResponse(BaseModel):
    order_id: str
    status: str
    total_amount: int


class GetOrdersResponse(BaseModel):
    orders: list[OrderSummaryResponse]
    total_count: int


class GetOrderResponse(BaseModel):
    order_id: str
    status: str
    total_amount: int

File: implementations/kotlin-springboot/examples/src/main/kotlin/com/example/orderservice/order/interfaces/rest/Schemas.kt
package com.example.orderservice.order.interfaces.rest

data class CreateOrderRequest(
    val userId: String,
    val items: List<ItemInput>,
) {
    data class ItemInput(val itemId: Int, val name: String, val price: Int, val quantity: Int)
}

data class CancelOrderRequest(val reason: String)

data class ErrorResponse(val message: String)
